### PR TITLE
Add type annotations and complete missing docstrings

### PR DESCRIPTION
**Pull request by brgcode**

### CODE REVIEW

File: src/compas_cgal/meshing.py [hunk 1]
```diff
@@ -1,15 +1,21 @@
+from typing import List, Tuple
 import numpy as np
 from compas_cgal._cgal import meshing
 from compas.plugins import plugin
 
+from compas_cgal.types import VerticesFaces, VerticesFacesNumpy
```
Comment (gonzalocasas) [inline src/compas_cgal/meshing.py:6]: I think this file is missing a lint/isort to fix the import style (force one line)

File: src/compas_cgal/meshing.py [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
     'remesh'
 ]
 
 
 @plugin(category='trimesh', pluggable_name='trimesh_remesh')
-def remesh(mesh, target_edge_length, number_of_iterations=10, do_project=True):
+def remesh(mesh: VerticesFaces,
+           target_edge_length: float,
+           number_of_iterations: int = 10,
+           do_project: bool = True) -> VerticesFacesNumpy:
```
Comment (gonzalocasas) [inline src/compas_cgal/meshing.py:18]: isn't it `bool=True` and `int=10` without spaces around `=`?

File: src/compas_cgal/trimesh.py [hunk 3]
```diff
@@ -1,5 +1,11 @@
+from __future__ import annotations
+from typing import Any, Dict, Optional, List, Tuple
```
Comment (gonzalocasas) [inline src/compas_cgal/trimesh.py:2]: same here, isort should prolly move this to separate lines

**Review by gonzalocasas — APPROVED** [review]
besides some small style details, LGTM!

File: src/compas_cgal/meshing.py (hunk 2 above)
Comment (brgcode) [reply src/compas_cgal/meshing.py:18]: apparently not in the case of using type annotations

**Pull request merged**